tests/acceptance/util.py

```diff
@@ -334,14 +334,19 @@ def wait_for_output_inner(
     ):
         output = proc.stdout.readline()
         while 1:
-            if re.search(expected_output, output.decode("utf8")):
-                break
+            decoded = output.decode("utf8")
+            match = re.search(expected_output, decoded)
+            if match:
+                return match
             for forbidden_output in forbidden_outputs:
-                assert not re.search(forbidden_output, output.decode("utf8")), output
+                assert not re.search(forbidden_output, decoded), output
             output = proc.stdout.readline()
 
-    wait_for_output_inner(proc, expected_output, forbidden_outputs)
+    match: re.Match[str] = cast(
+        re.Match[str], wait_for_output_inner(proc, expected_output, forbidden_outputs)
+    )
```

Comment:
I think it should be sufficient to declare the return type on `wait_for_output_inner`. Then you neither need the cast nor the type on the variable